### PR TITLE
Switch to GitHub Actions / GitHub Packages

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

TODO:

- [x] Confirm that pushing the images actually works (confirmed here: https://github.com/teohhanhui/api-platform/runs/326314060#step:15:1)
- [x] Push varnish image (from docker-compose-prod)
- [ ] Add container security vulnerability scanner? e.g. https://github.com/aquasecurity/trivy (because we lose this from quay.io which uses Clair)

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+          docker create --name docker_compose "docker/compose:${DOCKER_COMPOSE_VERSION}-alpine"
+          docker cp docker_compose:/usr/local/bin/docker-compose /usr/local/bin/docker-compose
+          docker rm docker_compose
+      - name: Fetch production setup
+        run: wget -O - https://github.com/api-platform/docker-compose-prod/archive/master.tar.gz | tar -xzf - && mv docker-compose-prod-master docker-compose-prod
+      - name: Pull images
+        run: docker-compose -f docker-compose-prod/docker-compose.build.yml pull --ignore-pull-failures
+      - name: Build images
+        run: |
+          docker-compose -f docker-compose-prod/docker-compose.build.yml config > docker-compose.bake.yml
```
Comment (teohhanhui) [inline .github/workflows/ci-cd.yml]: See https://github.com/docker/buildx/issues/197

**Comment by teohhanhui** [comment]
See jobs at https://github.com/teohhanhui/api-platform/runs/326304601

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        env:
+          DOCKER_COMPOSE_VERSION: '1.25.0'
+        run: |
+          docker create --name docker_compose "docker/compose:${DOCKER_COMPOSE_VERSION}-alpine"
+          docker cp docker_compose:/usr/local/bin/docker-compose /usr/local/bin/docker-compose
+          docker rm docker_compose
+      - name: Pull images
+        run: docker-compose pull --ignore-pull-failures
+      - name: Build images
+        run: docker buildx bake --pull
```
Comment (bendavies) [inline .github/workflows/ci-cd.yml]: you need to authenticate first for caching to work? 
https://github.com/api-platform/api-platform/commit/b15589d4543b75130ddbfb23578dc909d113e4dd/checks?check_suite_id=337568056#step:8:5

Comment (teohhanhui) [reply .github/workflows/ci-cd.yml]: I don't think so. GitHub Packages doesn't seem to support cache manifests.

**Pull request merged**